src/Model/Scope/CompoundCondition.php

```diff
@@ -69,17 +69,13 @@ public function __clone()
     }
 
     /**
-     * Add nested condition.
-     *
-     * @param mixed $field
-     * @param mixed $operator
-     * @param mixed $value
-     *
      * @return static
      */
-    public function addCondition($field, $operator = null, $value = null)
+    public function addCondition($key, $operator = null, $value = null)
     {
-        $this->add(new self([func_get_args()]));
```

Comment:
I would only change the statement to
`$this->add(static::createAnd(func_get_args()));`

Why would you want to remove the flexibility? It does create what you expect + more and consistent with `Model::addCondition`. And code is simpler.

Reply:
> Why would you want to remove the flexibility? It does create what you expect + more and consistent with `Model::addCondition`. And code is also simpler and all handling is in one place (`__constructor`).

> We can even use same method in `Model::addCondition`
> `$this->scope()->addCondition(...func_get_args());`

I like to think in consistent design.

There are two things:

a) I do not think it is good to wrap when we add a simple condition. The structure should really look like to what you add, if you add `name = "George"`, then I would never expect a Compound condition from it. And, this was also not the case before (in case of array structure). We can agree on this, right?

b) Allow to pass a conditions as an array, ie. `addCondition([[1. cond], [2. cond]])`, that is your concern, right?

If the input is an array, we can use CompoundCondition - but what is the usecase vs. not using createAnd/createOr which I think is easier to read, because:
- of explicit And/Or in text
- you can still add CompoundCondition object using this method
- we enforce the user to use object as soon as possible, ie. to not create an array first

Reply:
I have several concerns:
- Model will use addCondition in one way (BC considerations) and CompoundCondition in another (inconsistent)
- handling of creation of BasicCondition will be distributed on several places instead of only one (in constructor). When you addCondition you expect a condition to be added to the scope. And it is. in what way is internal workings.
- we reduce the flexibility
- we spend way too much time on theoretical discussions. At some point benefits of collaborative creation are wiped out as we cannot go further and just go in circles